misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/RangeFormatterTest.java

```diff
@@ -38,46 +44,32 @@
 /**
  * The abstract class that is extended by all range formatting test classes.
  *
- * @since 1.2.10
+ * @since 2.0.0
  */
-public abstract class RangeFormatter {
+public abstract class RangeFormatterTest {
     private final Path resourceDirectory = Paths.get("src").resolve("test").resolve("resources").toAbsolutePath();
-    private Path buildDirectory = Paths.get("build").toAbsolutePath().normalize();
     private static final String ASSERT_DIR = "assert";
     private static final String SOURCE_DIR = "source";
 
+    private static final Gson gson = new Gson();
+
     @Test(dataProvider = "test-file-provider")
-    public void test(String source, int[][] positions)
+    public void test(String sourcePath, String source, JsonArray positions)
             throws IOException, FormatterException {
-        Path assertFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), ASSERT_DIR, source);
-        Path sourceFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), SOURCE_DIR, source);
-
-        String content = getSourceText(sourceFilePath);
+        Path assertFilePath = Paths.get(resourceDirectory.toString(), sourcePath, ASSERT_DIR, source);
+        Path sourceFilePath = Paths.get(resourceDirectory.toString(), sourcePath, SOURCE_DIR, source);
+        String content = getFileContent(sourceFilePath);
         TextDocument textDocument = TextDocuments.from(content);
         SyntaxTree syntaxTree = SyntaxTree.from(textDocument, sourceFilePath.toString());
-        for (int[] position : positions) {
-            LinePosition startPos = LinePosition.from(position[0], position[1]);
-            LinePosition endPos = LinePosition.from(position[2], position[3]);
-            LineRange lineRange = LineRange.from(null, startPos, endPos);
+        for (JsonElement position : positions) {
+            JsonObject start = position.getAsJsonObject().get("startPos").getAsJsonObject();
+            LinePosition startPos = LinePosition.from(start.get("lineNo").getAsInt(), start.get("colNo").getAsInt());
+            JsonObject end = position.getAsJsonObject().get("endPos").getAsJsonObject();
+            LinePosition endPos = LinePosition.from(end.get("lineNo").getAsInt(), end.get("colNo").getAsInt());
+            LineRange lineRange = LineRange.from(sourceFilePath.toString(), startPos, endPos);
```

Comment:
Converting JSON to  `LineRange` is not part of the test case. Can we do this calculation at the data-provider's side?

Reply:
Moved to the data-provider's side. Now it's being calculated in the `getConfigsList()` method.